staging/provisioning/windows/hostsconfigagent.ps1

```diff
@@ -0,0 +1,54 @@
+$Global:ClusterConfiguration = ConvertFrom-Json ((Get-Content "c:\k\kubeclusterconfig.json" -ErrorAction Stop) | out-string)
+$clusterFQDN = $Global:ClusterConfiguration.Kubernetes.ControlPlane.IpAddress
+$hostsFile="C:\Windows\System32\drivers\etc\hosts"
+$retryDelaySeconds = 15
+
+# Dot-source scripts with functions that are called in this script
+. c:\AzureData\k8s\kuberneteswindowsfunctions.ps1
+
+function Get-APIServer-IPAddress
+{
+    $uri = "http://169.254.169.254/metadata/instance/compute/tags?api-version=2019-03-11&format=text"
+    $response = Retry-Command -Command "Invoke-RestMethod" -Args @{Uri=$uri; Method="Get"; ContentType="application/json"; Headers=@{"Metadata"="true"}} -Retries 3 -RetryDelaySeconds 5
+
+    if(!$response) {
+        return ""
+    }
+
+    foreach ($tag in $response.Split(";"))
+    {
+        $values = $tag.Split(":")
+        if ($values.Length -ne 2)
+        {
+            return ""
+        }
+
+        if ($values[0] -eq "aksAPIServerIPAddress")
+        {
+            return $values[1]
+        }
+    }
+
+    return ""
+}
+
+Write-Log "Get cluster APIServer FQDN: $clusterFQDN"
+while ($true)
+{
+    $clusterIP = Get-APIServer-IPAddress
+    if ($clusterIP -eq "") {
+        Start-Sleep $retryDelaySeconds
+        continue
+    }
+
+    $hostsContent=Get-Content -Path $hostsFile -Encoding UTF8
+    if ($hostsContent -match "$clusterIP $clusterFQDN") {
+        Write-Log "APIServer FQDN has already been set to $clusterIP in hosts file"
+    } else {
+        $hostsContent -notmatch "$clusterFQDN" | Out-File $hostsFile -Encoding UTF8
+        Add-Content -Path $hostsFile -Value "$clusterIP $clusterFQDN" -Encoding UTF8
```

Comment:
This appends to the host file. If the value changes do you need to clean up the old one?

Reply:
it is clean up in the above line

Reply:
I don't have enough context. It looks like the [linux solution](https://github.com/Azure/aks-engine/pull/3556/files) is doing the same so I say we keep similar